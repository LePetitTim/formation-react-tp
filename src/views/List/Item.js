import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';

import { removeItem } from '../../services/store/actions';

export class Item extends React.Component {
  static propTypes = {
    id: PropTypes.number,
    title: PropTypes.string,
  };

  remove = async () => {
    const { id, removeItem } = this.props;
    removeItem(id);
  }

  render () {
    const { id, title } = this.props;
    const { remove } = this;

    return (
      <tr>
        <td>{id}</td>
        <td>
          <NavLink
            to={`/edit/${id}`}>
            {title}
          </NavLink>
        </td>
        <td>
          <button
            type="button"
            onClick={remove}>Supprimer</button>
        </td>
      </tr>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  removeItem: id => dispatch(removeItem(id)),
});

export default connect(null, mapDispatchToProps)(Item);
