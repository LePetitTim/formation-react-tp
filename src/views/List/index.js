import React from 'react';
import { connect } from 'react-redux';

import { getAllItems } from '../../services/store/actions';

import Loading from '../../components/Loading';
import Item from './Item';
export class List extends React.Component {
  componentDidMount() {
    this.props.getAll();
  }

  render () {
    const { loading, items } = this.props;
    const { onRemove } = this;
    return (loading && !items.length)
      ? (
        <Loading />
      ) : (
        <table>
          <thead>
            <tr>
              <td>ID</td>
              <td>Titre</td>
              <td>Actions</td>
            </tr>
          </thead>
          <tbody>
            {items.map(item => (
              <Item
                key={item.id}
                {...item}
                onRemove={onRemove} />
            ))}
          </tbody>
        </table>
      )
  }
}

const mapStateToProps = state => ({
  items: state.items,
  loading: state.loading,
});

const mapDispatchToProps = dispatch => ({
  getAll: () => dispatch(getAllItems()),
});

export default connect(mapStateToProps, mapDispatchToProps)(List);
