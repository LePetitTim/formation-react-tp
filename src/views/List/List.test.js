import React from 'react';
import renderer from 'react-test-renderer';
import List from './';

it('should render', () => {
  const tree = renderer.create(<List />).toJSON();
  expect(tree).toMatchSnapshot();
});
