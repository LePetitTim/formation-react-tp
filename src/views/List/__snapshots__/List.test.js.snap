// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<table>
  <thead>
    <tr>
      <td>
        ID
      </td>
      <td>
        Titre
      </td>
      <td>
        Actions
      </td>
    </tr>
  </thead>
  <tbody />
</table>
`;
